esphome/components/ld2410s/LD2410S.cpp

```diff
@@ -0,0 +1,947 @@
+  this->schedule_cmd_frame_(FW_READ_CMD);
+  this->schedule_cmd_frame_(PARAMS_READ_CMD);
+  this->schedule_cmd_frame_(GATE_TRIGGER_THRESHOLD_READ_CMD);
+  this->schedule_cmd_frame_(GATE_HOLD_THRESHOLD_READ_CMD);
+  this->schedule_cmd_frame_(GATE_SNR_READ_CMD);
+  this->schedule_cmd_frame_(CONFIG_MODE_END_CMD);
+
+  this->status_clear_warning();
+}
+void LD2410S::write_all() {
+  this->status_set_warning("write_all");
+  this->schedule_cmd_frame_(CONFIG_MODE_START_CMD);
+
+  this->schedule_cmd_frame_(OUTPUT_MODE_SWITCH_CMD);
+  this->schedule_cmd_frame_(PARAMS_WRITE_CMD);
+  this->schedule_cmd_frame_(GATE_TRIGGER_THRESHOLD_WRITE_CMD);
+  this->schedule_cmd_frame_(GATE_HOLD_THRESHOLD_WRITE_CMD);
+  this->schedule_cmd_frame_(GATE_SNR_WRITE_CMD);
+
+  this->schedule_cmd_frame_(PARAMS_READ_CMD);
+  this->schedule_cmd_frame_(GATE_TRIGGER_THRESHOLD_READ_CMD);
+  this->schedule_cmd_frame_(GATE_HOLD_THRESHOLD_READ_CMD);
+  this->schedule_cmd_frame_(GATE_SNR_READ_CMD);
+
+  this->schedule_cmd_frame_(CONFIG_MODE_END_CMD);
+  this->status_clear_warning();
+}
+void LD2410S::calibration() { this->schedule_cmd_("start_calibration\0", CALIBRATION_CMD); }
+void LD2410S::factory_reset() {
+  this->minimal_output_ = true;
+
+  this->max_dist_ = 16;
+  this->min_dist_ = 0;
+  this->delay_ = 10;
+  this->status_freq_ = 80;
+  this->dist_freq_ = 80;
+  this->resp_speed_ = 5;
+
+  for (int i = 0; i < 16; i++) {
+    this->triggers_.trigger[i] = GATE_TRIGGER_THRESHOLD_WRITE_DATA[i];
+    this->triggers_.hold[i] = GATE_HOLD_THRESHOLD_WRITE_DATA[i];
+    this->triggers_.snr[i] = GATE_SNR_WRITE_DATA[i];
+  }
+
+  this->status_set_warning("factory_reset");
+  this->schedule_cmd_frame_(CONFIG_MODE_START_CMD);
+
+  this->schedule_cmd_frame_(OUTPUT_MODE_SWITCH_CMD);
+
+  this->schedule_cmd_frame_(PARAMS_WRITE_CMD);
+  this->schedule_cmd_frame_(GATE_TRIGGER_THRESHOLD_WRITE_CMD);
+  this->schedule_cmd_frame_(GATE_HOLD_THRESHOLD_WRITE_CMD);
+  this->schedule_cmd_frame_(GATE_SNR_WRITE_CMD);
+
+  this->schedule_cmd_frame_(PARAMS_READ_CMD);
+  this->schedule_cmd_frame_(GATE_TRIGGER_THRESHOLD_READ_CMD);
+  this->schedule_cmd_frame_(GATE_HOLD_THRESHOLD_READ_CMD);
+  this->schedule_cmd_frame_(GATE_SNR_READ_CMD);
+
+  this->schedule_cmd_frame_(CONFIG_MODE_END_CMD);
+  this->status_clear_warning();
+}
+void LD2410S::toggle_minimal() {
+  this->minimal_output_ = !this->minimal_output_;
+  this->schedule_cmd_("toggle_minimal_output\0", OUTPUT_MODE_SWITCH_CMD);
+}
+
+void LD2410S::set_delay(float delay) {
+  this->delay_ = delay;
+  this->schedule_cmd_("set_delay\0", PARAMS_WRITE_CMD, CFG_NO_DELAY_VALUE);
+}
+void LD2410S::set_distance_reporting_freq(float distance_reporting_freq) {
+  this->dist_freq_ = distance_reporting_freq * 10;
+  this->schedule_cmd_("set_distance_reporting_freq\0", PARAMS_WRITE_CMD, CFG_DISTANCE_FREQ_VALUE);
+}
+void LD2410S::set_max_distance(float max_distance) {
+  this->max_dist_ = static_cast<float>(max_distance) / 0.7;
```

Comment:
```suggestion
  this->max_dist_ = static_cast<float>(max_distance) / 0.7f;
```
...otherwise the computation is done as `double` in software instead of hardware.
I see this again a few lines down, too.